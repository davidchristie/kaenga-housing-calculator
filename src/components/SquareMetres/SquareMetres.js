/* eslint-disable react/style-prop-object */

import React, { Component, PropTypes } from 'react'
import { FormattedNumber } from 'react-intl'

class SquareMetres extends Component {
  render () {
    const { value } = this.props
    return (
      <span>
        <FormattedNumber
          maximumFractionDigits={1}
          value={value}
        />m<sup>2</sup>
      </span>
    )
  }
}

SquareMetres.defaultProps = {
  value: 0
}

SquareMetres.propTypes = {
  value: PropTypes.number.isRequired
}

export default SquareMetres
