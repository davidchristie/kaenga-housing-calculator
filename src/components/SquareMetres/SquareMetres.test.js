import React from 'react'
import ReactTestUtils from 'react-addons-test-utils'

import SquareMetres from './SquareMetres'

const shallowRenderer = ReactTestUtils.createRenderer()

it('matches snapshot', () => {
  expect(shallowRenderer.render(<SquareMetres />)).toMatchSnapshot()
})
