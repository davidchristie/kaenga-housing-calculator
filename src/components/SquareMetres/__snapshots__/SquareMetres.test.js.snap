exports[`test matches snapshot 1`] = `
<span>
  <FormattedNumber
    maximumFractionDigits={1}
    value={0} />
  m
  <sup>
    2
  </sup>
</span>
`;
